packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -119,6 +136,27 @@ class StateMachineBloc {
     }
   }
 
+  /// Listens for asynchronous events which occurred outside the control of the
+  /// [Authenticator] and [StateMachineBloc].
+  AuthState? _mapHubEvent(AuthHubEvent event) {
+    logger.debug('Handling hub event: ${event.type}');
+    switch (event.type) {
+      case AuthHubEventType.signedIn:
+        if (_currentState is! AuthenticatedState) {
+          return const AuthenticatedState();
+        }
+        break;
+      case AuthHubEventType.signedOut:
+      case AuthHubEventType.sessionExpired:
```

Comment:
When a user's session expires they will be taken to the sign in screen. We will not provide any message to them explaining why. The assumption is that the customer will be responsible for listening to the hub events and providing some sort of info in the UI if desired... Will they be able to do that easily?

I think the UX is fine if this happens on app launch, but it is a poor UX if it happens while the user is in the middle of something. Should we attempt to show something in the UI in this case?

I think this could be a follow up. I suspect we already are not handling this case well. Can we create a task for it if we think this should be improved?

Reply:
Done 👍 good callout